### PR TITLE
Remove server renderer

### PR DESCRIPTION
**Pull request by alan-agius4**
- See individual commits
- Dependent on: https://github.com/angular/domino/pull/14
- TGP 🟢: http://test/OCL:521669156:BASE:521703873:1680603082207:299be4a4

### CODE REVIEW

File: integration/cli-elements-universal/package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "lint": "ng lint",
     "e2e": "ng e2e --port 0",
     "pretest-ssr": "yarn ng run cli-elements-universal:app-shell:production",
-    "test-ssr": "node --eval \"assert(fs.readFileSync('dist/cli-elements-universal/browser/index.html', 'utf8').includes('<app-title-ce>'), 'Expected \\'index.html\\' to contain \\'<app-title-ce>\\'.');\""
+    "test-ssr": "node --eval \"assert(fs.readFileSync('dist/cli-elements-universal/browser/index.html', 'utf8').includes('<app-title-ce appname'), 'Expected \\'index.html\\' to contain \\'<app-title-ce appname\\'.');\""
```
Comment (alan-agius4) [inline integration/cli-elements-universal/package.json:13]: The test here expect was actually incorrect.

File: packages/platform-server/test/integration_spec.ts [hunk 2]
```diff
@@ -709,7 +709,7 @@ describe('platform-server integration', () => {
     let doc: string;
     let called: boolean;
     let expectedOutput =
-        '<html><head></head><body><app ng-version="0.0.0-PLACEHOLDER" ng-server-context="other">Works!<h1 textcontent="fine">fine</h1></app></body></html>';
+        '<html><head></head><body><app ng-version="0.0.0-PLACEHOLDER" ng-server-context="other">Works!<h1>fine</h1></app></body></html>';
```
Comment (alan-agius4) [inline packages/platform-server/test/integration_spec.ts:712]: The expects here were actually incorrect.

**Review by JiaLiPassion — APPROVED** [review]
LGTM!
reviewed-for: fw-zones

**Review by JiaLiPassion — APPROVED** [review]
LGTM!
reviewed-for: zone-js

**Review by jessicajaniuk — APPROVED** [review]
LGTM

reviewed-for: fw-animations, fw-core, fw-platform-server, integration-tests

File: packages/animations/browser/src/render/shared.ts [hunk 3]
```diff
@@ -140,51 +122,17 @@ export function parseTimelineCommand(command: string): [string, string] {
...
-      return [];
-    };
-let _documentElement: unknown|null = null;
+const documentElement: HTMLElement|null =
+    /* @__PURE__ */ (() => typeof document === 'undefined' ? null : document.documentElement)();
 
 export function getParentElement(element: any): unknown|null {
   const parent = element.parentNode || element.host;  // consider host to support shadow DOM
-  if (parent === _documentElement) {
+  if (parent == documentElement) {
```
Comment (AndrewKushnir) [inline packages/animations/browser/src/render/shared.ts:130]: Is it important to have `==` here (vs `===`)?
```suggestion
  if (parent === documentElement) {
```

File: packages/platform-browser/src/dom/dom_renderer.ts [hunk 4]
```diff
@@ -278,6 +274,11 @@ class DefaultDomRenderer2 implements Renderer2 {
 
   setProperty(el: any, name: string, value: any): void {
     (typeof ngDevMode === 'undefined' || ngDevMode) && checkNoSyntheticProp(name, 'property');
+    // Mirror property as Domino has limited property access.
+    if (this.platformIsServer && !dominoElementHasProperty(el, name)) {
+      this.setAttribute(el, name, value);
+    }
```
Comment (AndrewKushnir) [inline packages/platform-browser/src/dom/dom_renderer.ts:280]: What'd be an example of such property? Just curious what happens in a browser in this case (if we are consistent).

File: packages/platform-server/src/server.ts [hunk 5]
```diff
@@ -42,18 +40,7 @@ function initDominoAdapter() {
...
-  return new ɵAnimationRendererFactory(renderer, engine, zone);
-}
-
 export const SERVER_RENDER_PROVIDERS: Provider[] = [
-  ServerRendererFactory2,
-  {
-    provide: RendererFactory2,
-    useFactory: instantiateServerRendererFactory,
-    deps: [ServerRendererFactory2, ɵAnimationEngine, NgZone]
-  },
```
Comment (AndrewKushnir) [inline packages/platform-server/src/server.ts:56]: 🎉

**Review by AndrewKushnir — COMMENTED** [review]
LGTM, just a couple comments. Thanks Alan!

**Comment by AndrewKushnir** [comment]
**Caretaker note** (for additional context): this PR can land **after the feature freeze**, since this is a refactoring. However we want to land it before v16 RC.0, thus it's in the `v16-candidates` milestone.

**Comment by AndrewKushnir** [comment]
FYI, removed the PR from the `v16-candidates` milestone for now, so we focus on the remaining PR that need to land before the feature freeze.

**Comment by alan-agius4** [comment]
TGP: http://test/OCL:521669156:BASE:522015502:1680694242942:62500b20

File: packages/platform-browser/src/dom/dom_renderer.ts [hunk 6]
```diff
@@ -57,52 +57,39 @@ export function shimStylesContent(compId: string, styles: string[]): string[] {
...
   }
 
   createRenderer(element: any, type: RendererType2|null): Renderer2 {
     if (!element || !type) {
       return this.defaultRenderer;
     }
 
-    const renderer = this.getOrCreateRenderer(element, type);
+    if (this.platformIsServer && type.encapsulation === ViewEncapsulation.ShadowDom) {
+      // Domino does not support shadow DOM.
```
Comment (AndrewKushnir) [inline packages/platform-browser/src/dom/dom_renderer.ts:88]: nit: I think it'd be great to add a comment here to explain **why it's safe** to change the `encapsulation` to `Emulated`.

**Review by AndrewKushnir — APPROVED** [review]
Looks great, thanks Alan! 👍

File: packages/platform-browser/src/dom/dom_renderer.ts (hunk 6 above)
Comment (alan-agius4) [reply packages/platform-browser/src/dom/dom_renderer.ts:88]: Let me do that in a follow up

**Pull request closed without merging**